python/src/trezorlib/cardano.py

```diff
@@ -161,6 +201,9 @@ def sign_tx(
     outputs: List[messages.CardanoTxOutputType],
     fee: int,
     ttl: int,
+    certificates: List[messages.CardanoTxCertificateType],
```

Comment:
please use default values for these fields: presumably `()` for certificates and withdrawals (although i'm currently unsure if our protobuf will correctly accept that instead of a list; might be necessary to use a `None` value and replace with an empty list in the function body,

`None` for metadata hash?

fwiw maybe `ttl`, `protocol_magic` and `network_id` should also have default values?

Reply:
Added [here](https://github.com/trezor/trezor-firmware/pull/1145/commits/ff25224d44cdb9113a87b204e344b2ae5d403cf0). I've also tested it by removing the parameters from cli and checking the types and it indeed seemed to convert it to lists - is that a sufficient test or should I rather use `None` and replace it? ... Why can't we actually use `[]`?

Reply:
> Why can't we actually use `[]`?

Because that list is the same one for the life of the python interpreter, so if anyone were to somehow add items into it, these items would then always be sent.
It might be technically safe in this case, but it's error prone and some linters _will_ shout about it.

if you tried it and it worked, then let's leave it like this

Reply:
Oh, thank you for the explanation!